File: java/calendar/src/test/java/com/rbleggi/calendar/MeetingTest.java
package com.rbleggi.calendar;

import org.junit.jupiter.api.Test;
import java.time.LocalDateTime;
import java.util.Set;
import static org.junit.jupiter.api.Assertions.*;

class MeetingTest {
    @Test
    void meetingHasCorrectTitle() {
        var user = new User("1", "Alice");
        var meeting = new Meeting("m1", "Sync", LocalDateTime.now(), LocalDateTime.now().plusHours(1), Set.of(user));
        assertEquals("Sync", meeting.title());
    }

    @Test
    void meetingHasCorrectId() {
        var user = new User("1", "Alice");
        var meeting = new Meeting("m1", "Sync", LocalDateTime.now(), LocalDateTime.now().plusHours(1), Set.of(user));
        assertEquals("m1", meeting.id());
    }
}

File: kotlin/calendar/src/test/kotlin/com/rbleggi/calendar/MeetingTest.kt
package com.rbleggi.calendar

import java.time.LocalDateTime
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertNotNull

class MeetingTest {
    @Test
    fun `meeting has correct title`() {
        val user = User("1", "Alice")
        val meeting = Meeting(title = "Test", start = LocalDateTime.now(), end = LocalDateTime.now().plusHours(1), attendees = setOf(user))
        assertEquals("Test", meeting.title)
    }

    @Test
    fun `meeting generates id`() {
        val user = User("1", "Alice")
        val meeting = Meeting(title = "Test", start = LocalDateTime.now(), end = LocalDateTime.now().plusHours(1), attendees = setOf(user))
        assertNotNull(meeting.id)
    }
}
